lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/evaluator/Evaluators.java

```diff
@@ -53,7 +52,9 @@
   @Inject
   Evaluators() {
     LOG.log(Level.FINE, "Instantiated 'Evaluators'");
-    assert SingletonAsserter.assertSingleton(Evaluators.class);
+    // There can be several instances of the class for multiple REEFEnvironments.
+    // It is still a singleton when REEF Driver owns the entire JVM.
+    // assert SingletonAsserter.assertSingleton(Evaluators.class);
```

Comment:
Is it feasible to distinguish the two code paths? When we do and do not own the JVM? These assertions have been instrumental in debugging REEF and Tang.